### PR TITLE
[SYCL][NFC] Share the __has_builtin macro across the SYCL library

### PR DESCRIPTION
**Pull request by AGindinson**
This patch provides a common definition for a Clang-specific
macro and shares the definition among the current macro users.
This should improve the library user's code buildability by
3rd-party compilers.

Signed-off-by: Artem Gindinson <artem.gindinson@intel.com>

### CODE REVIEW

File: sycl/include/CL/sycl/detail/defines.hpp [hunk 1]
```diff
@@ -18,6 +18,10 @@
 #define __has_attribute(x) 0
 #endif
 
+#ifndef __has_builtin
```
Comment (bader) [inline sycl/include/CL/sycl/detail/defines.hpp]: Just for sake of readability, could you move this block above `__has_attribute` check?

Comment (AGindinson) [reply sycl/include/CL/sycl/detail/defines.hpp]: Moved it below instead to make stuff more alphabetical. Would this work as well?

Comment (bader) [reply sycl/include/CL/sycl/detail/defines.hpp]: Sure. Thanks.

**Comment by AGindinson** [comment]
The failure in `Clang-Unit::FormatTestComments.MultiLineCommentsInDefines` seems unrelated.

**Pull request merged**